src/cursor/cur_layered.c

```diff
@@ -1127,6 +1122,52 @@ __clayered_put(WT_SESSION_IMPL *session, WT_CURSOR_LAYERED *clayered, const WT_I
     return (0);
 }
 
+/*
+ * __clayered_remove_int --
+ *     Remove an entry from the desired tree.
+ */
+static WT_INLINE int
+__clayered_remove_int(
+  WT_SESSION_IMPL *session, WT_CURSOR_LAYERED *clayered, const WT_ITEM *key, bool positioned)
+{
+    WT_CURSOR *c;
+
+    if (S2C(session)->layered_table_manager.leader) {
+        c = clayered->stable_cursor;
+        clayered->current_cursor = c;
+        if (!positioned) {
+            /*
+             * Clear the existing cursor position. Don't clear the primary cursor: we're about to
+             * use it anyway. We need the cursor still be positioned after the remove. Don't release
+             * the cursor if that is the case. Remove only retains the cursor position if it is
+             * positioned at the start.
+             */
+            WT_RET(__clayered_reset_cursors(clayered, true));
+            c->set_key(c, key);
+        } else
+            WT_ASSERT(session, F_ISSET(c, WT_CURSTD_KEY_INT));
+        WT_RET(c->remove(c));
+    } else {
+        c = clayered->ingest_cursor;
+        clayered->current_cursor = c;
+        if (!positioned) {
+            /*
+             * Clear the existing cursor position. Don't clear the primary cursor: we're about to
+             * use it anyway. No need to do another search if we are already positioned.
+             */
+            WT_RET(__clayered_reset_cursors(clayered, true));
+            c->set_key(c, key);
+        } else
+            WT_ASSERT(session, F_ISSET(c, WT_CURSTD_KEY_INT));
+        c->set_value(c, &__tombstone);
+        WT_RET(c->update(c));
+    }
+
+    /* TODO: Need something to add a log record? */
```

Comment:
Why would we need a log record?

Reply:
This is copied from __clayered_put. I think it means we need to log the record if logging enabled? I will remove this if logging is no longer relevant.